executor/simple.go

```diff
@@ -1067,13 +1074,51 @@ func (e *SimpleExec) executeSetPwd(s *ast.SetPwdStmt) error {
 	return err
 }
 
-func (e *SimpleExec) executeKillStmt(s *ast.KillStmt) error {
+func (e *SimpleExec) executeKillStmt(ctx context.Context, s *ast.KillStmt) error {
+	sm := e.ctx.GetSessionManager()
+	if sm == nil {
+		return nil
+	}
+
+	if e.InCoprocessor {
+		logutil.BgLogger().Info("KILL in coprocessor", zap.Uint64("connID", s.ConnectionID), zap.Bool("query", s.Query),
+			zap.String("sourceAddr", e.ctx.GetSessionVars().SourceAddr.IP.String()))
+		sm.Kill(s.ConnectionID, s.Query)
+		return nil
+	}
+
+	if sm.ServerID() == 0 {
+		// `sm.ServerID() == 0` indicates serverID allocation would be not working.
+		// Degrade to behavior as early versions.
+		return e.executeCompatibleKill(ctx, s)
```

Comment:
Could we drop this because there're no conflicting connection IDs by design after this PR?